File: pinyougou_partent_project/pinyougou_manager_web/src/main/webapp/js/service/brandService.js
//服务分层
app.service('brandService', function ($http) {
    //查询品牌列表集合
    this.selectOptionList=function () {
        return $http.get("../brand/selectOptionList.do")
    };
    this.findByPage = function (currPage, pageSize) {
        return $http.get('../brand/findByPage.do?currPage=' + currPage + '&pageSize=' + pageSize);
    };
    this.findOne = function (id) {
        return $http.get('../brand/findOne.do?id=' + id);
    };
    this.add = function (entity) {
        return $http.post('../brand/add.do', entity);
    };
    this.updateBrand = function (entity) {
        return $http.post('../brand/updateBrand.do', entity);
    };
    this.deleteBrand = function (ids) {
        return $http.post('../brand/deleteBrand.do?ids=' + ids)
    };
    this.search = function (currPage, pageSize, searchEntity) {
        return $http.post('../brand/findBrandByCondition.do?currPage=' + currPage + '&pageSize=' + pageSize, searchEntity)
    };
});

File: pinyougou_partent_project/pinyougou_sellergoods_interface/src/main/java/com/pinyougou/service/BrandService.java
package com.pinyougou.service;

import com.pinyougou.page.PageResult;
import com.pinyougou.pojo.TbBrand;

import java.util.List;
import java.util.Map;

public interface BrandService {
    /**
     * 查询所有
     * @return
     */
    public List<TbBrand> findAll();

    /**
     * 分页查询
     * @param currPage
     * @param pageSize
     * @return
     */
    PageResult<TbBrand> findByPage(Integer currPage, Integer pageSize);

    /**
     * 品牌添加
     * @param tbBrand
     */
    void add(TbBrand tbBrand);

    /**
     * id品牌查找
     * @param id
     * @return
     */
    TbBrand findOne(Long id);

    /**
     * 品牌修改
     * @param tbBrand
     */
    void updateBrand(TbBrand tbBrand);

    /**
     * 品牌删除
     * @param ids
     */
    void deleteBrand(Long[] ids);

    /**
     * 条件查找
     * @param tbBrand
     * @return
     */
    PageResult findBrandByCondition(TbBrand tbBrand,Integer currPage,Integer pageSize);

    /**
     * 查询品牌列表集合
     * @return
     */
    List<Map> selectOptionList();
}
